### PR TITLE
Intel Hex format unpacker/packer

### PR DESCRIPTION
**Pull request by EdwardLarson**
- [x] I have reviewed the [OFRAK contributor guide](https://ofrak.com/docs/contributor-guide/getting-started.html) and attest that this pull request is in accordance with it.

**One sentence summary of this PR (This should go in the CHANGELOG!)**
Add Identifier, Unpacker, Packer for Intel Hex format

**Link to Related Issue(s)**

**Please describe the changes in your request.**

**Anyone you think should look at this, specifically?**

### CODE REVIEW

File: ofrak_core/ofrak/core/ihex.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
+import logging
+import re
+import sys
+from dataclasses import dataclass
+from typing import List, Union, Tuple
+
+from bincopy import BinFile  # type:ignore
```
Comment (rbs-jacob) [inline ofrak_core/ofrak/core/ihex.py:7]: Out of curiosity, why does this need a type ignore? I haven't noticed imports causing mypy errors before.

File: ofrak_core/ofrak/core/ihex.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    async def unpack(self, resource: Resource, config=None):
+        if resource.has_tag(Ihex):
+            ihex_program, binfile = _binfile_analysis(await resource.get_data())
+
+            await resource.create_child_from_view(ihex_program, data=bytes(binfile.as_binary()))
+
+        elif resource.has_tag(IhexProgram):
+            ihex_program = await resource.view_as(IhexProgram)
+            for seg in ihex_program.segments:
+                segment_data_range = seg.translate(-ihex_program.address_limits.start)
```
Comment (rbs-jacob) [inline ofrak_core/ofrak/core/ihex.py:74]: Why this translation? An explanatory comment might be helpful.

**Review by rbs-jacob — CHANGES_REQUESTED** [review]
It probably makes sense to change the packer and unpacker to become two packers and two unpackers. 

Right now both have two distinct code paths that case on tags. This is exactly what separate components are for, and I don't see the benefit to having those if statements each be inside one packer/unpacker instead of having the cases broken up across different packers and unpackers.

File: ofrak_core/ofrak/core/ihex.py (hunk 1 above)
Comment (EdwardLarson) [reply ofrak_core/ofrak/core/ihex.py]: It was to deal with the mypy error about not being able to type check it:
> ofrak/core/ihex.py:7: error: Skipping analyzing "bincopy": module is installed, but missing library stubs or py.typed marker

The way we usually deal with them is adding an ignore for that module in the `mypy.ini` file. Updated.

**Review by rbs-jacob — APPROVED** [review]

**Pull request merged**